lib/ember-test-helpers/test-module-for-component.js

```diff
@@ -190,6 +190,70 @@ export default TestModule.extend({
     };
   },
 
+  setupComponentLegacyIntegrationTest: function () {
+    var resolver = getResolver();
+    var namespace = Ember.Object.create({
+      Resolver: { create: function() { return resolver; } }
+    });
+
+    if (Ember.Application.buildRegistry) {
+      var registry;
+      registry = Ember.Application.buildRegistry(namespace);
+      registry.register('component-lookup:main', Ember.ComponentLookup);
+      this.registry = registry;
+      this.container = registry.container();
+    } else {
+      this.container = Ember.Application.buildContainer(namespace);
+      this.container.register('component-lookup:main', Ember.ComponentLookup);
+    }
+  },
+
+  setupLegacyComponentTest: function () {
```

Comment:
Sorry for beating around the bush so much here.  I think that legacy mode should do the following:

* Change the conditional https://github.com/switchfly/ember-test-helpers/blob/master/lib/ember-test-helpers/test-module.js#L129-L135 to just use `_setupIntegratedContainer` for legacy mode.
* Use `setupComponentUnitTest` when in legacy mode.

The combination of those two changes, essentially sets us back to the behavior of ember-test-helpers 0.3.6 - 0.4.3.